addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-gcp/src/main/java/org/nuxeo/ecm/core/storage/gcp/GoogleStorageBinaryManager.java

```diff
@@ -0,0 +1,272 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nuxeo
+ */
+package org.nuxeo.ecm.core.storage.gcp;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.blob.AbstractBinaryGarbageCollector;
+import org.nuxeo.ecm.blob.AbstractCloudBinaryManager;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.blob.binary.BinaryGarbageCollector;
+import org.nuxeo.ecm.core.blob.binary.FileStorage;
+
+import com.google.api.gax.paging.Page;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.Storage.BlobField;
+import com.google.cloud.storage.Storage.BlobListOption;
+import com.google.cloud.storage.StorageOptions;
+
+/**
+ * A Binary Manager that stores binaries as Google Storage BLOBs
+ * <p>
+ * The BLOBs are cached locally on first access for efficiency.
+ * <p>
+ * Because the BLOB length can be accessed independently of the binary stream, it is also cached in a simple text file
+ * if accessed before the stream.
+ *
+ * @since 10.10-HF12
+ */
+public class GoogleStorageBinaryManager extends AbstractCloudBinaryManager {
+
+    private static final Log log = LogFactory.getLog(GoogleStorageBinaryManager.class);
+
+    public static final String BUCKET_NAME_PROPERTY = "storage.bucket";
+
+    public static final String BUCKET_PREFIX_PROPERTY = "storage.bucket_prefix";
+
+    public static final String PROJECT_ID_PROPERTY = "project";
+
+    public static final String GOOGLE_APPLICATION_CREDENTIALS = "credentials";
+
+    public static final String GOOGLE_PLATFORM_SCOPE = "https://www.googleapis.com/auth/cloud-platform";
+
+    public static final String GOOGLE_STORAGE_SCOPE = "https://www.googleapis.com/auth/devstorage.full_control";
+
+    public static final String SYSTEM_PROPERTY_PREFIX = "nuxeo.gcp";
+
+    private static final Pattern MD5_RE = Pattern.compile("[0-9a-f]{32}");
+
+    public static final String DELIMITER = "/";
+
+    protected String bucketName;
+
+    protected String bucketNamePrefix;
```

Comment:
Could you rename to `bucketPrefix`? The name `bucketNamePrefix` in the S3 connector is actually wrong and should be changed too.